constellation_base/views.py

```diff
@@ -57,3 +62,52 @@ def login_view(request):
 def logout_view(request):
     logout(request)
     return redirect("Login")
+
+
+@method_decorator(staff_member_required, name='dispatch')
+class manage_groups(View):
+
+    def get(self, request):
+        """Show a list of the supplemental groups"""
+        template_settings_object = GlobalTemplateSettings(allowBackground=True)
+        template_settings = template_settings_object.settings_dict()
+
+        existing_groups = LocalGroupACLEntry.objects.values_list('group',
+                                                                 flat=True)
+        existing_groups = Group.objects.filter(pk__in=set(existing_groups))
+
+        all_groups = Group.objects.all()
+
+        return render(request, 'constellation_base/group_list.html', {
+            'template_settings': template_settings,
+            'existing_groups': existing_groups,
+            'all_groups': all_groups
+        })
+
+    def post(self, request):
+
+        with transaction.atomic():
```

Comment:
try except allows you to handle integrity exceptions (see https://docs.djangoproject.com/en/1.11/topics/db/transactions/). It's a good idea imo